api/repositories/service_binding_repository_test.go

```diff
@@ -543,40 +554,61 @@ var _ = Describe("ServiceBindingRepo", func() {
 				).To(Succeed())
 
 				Expect(serviceBinding.Labels).To(HaveKeyWithValue("servicebinding.io/provisioned-service", "true"))
-				Expect(serviceBinding.Spec).To(Equal(
-					korifiv1alpha1.CFServiceBindingSpec{
-						DisplayName: nil,
-						Service: corev1.ObjectReference{
-							Kind:       "CFServiceInstance",
-							APIVersion: korifiv1alpha1.SchemeGroupVersion.Identifier(),
-							Name:       cfServiceInstance.Name,
-						},
-						AppRef: corev1.LocalObjectReference{
-							Name: appGUID,
-						},
-					},
-				))
+				Expect(serviceBinding.Spec.Type).To(Equal(korifiv1alpha1.CFServiceBindingTypeApp))
+				Expect(*(serviceBinding.Spec.DisplayName)).To(Equal(serviceBindingName))
+				Expect(serviceBinding.Spec.Service).To(Equal(corev1.ObjectReference{
+					Kind:       "CFServiceInstance",
+					APIVersion: korifiv1alpha1.SchemeGroupVersion.Identifier(),
+					Name:       cfServiceInstance.Name,
+				}))
+				Expect(serviceBinding.Spec.AppRef).To(Equal(corev1.LocalObjectReference{
+					Name: appGUID,
+				}))
 			})
+		})
 
-			When("the app does not exist", func() {
-				BeforeEach(func() {
-					appGUID = "i-do-not-exits"
-				})
+		When("the app does not exist", func() {
+			BeforeEach(func() {
+				appGUID = "i-do-not-exits"
+			})
 
-				It("reuturns an UnprocessableEntity error", func() {
-					Expect(createErr).To(BeAssignableToTypeOf(apierrors.UnprocessableEntityError{}))
-				})
+			It("returns an UnprocessableEntity error", func() {
+				Expect(createErr).To(BeAssignableToTypeOf(apierrors.UnprocessableEntityError{}))
 			})
+		})
 
-			When("The service binding has a name", func() {
-				BeforeEach(func() {
-					tempName := "some-name-for-a-binding"
-					bindingName = &tempName
-				})
+		When("binding type is key", func() {
+			BeforeEach(func() {
+				createMsg.Type = korifiv1alpha1.CFServiceBindingTypeKey
+				createMsg.AppGUID = ""
+				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, space.Name)
+			})
 
-				It("creates the binding with the specified name", func() {
-					Expect(serviceBindingRecord.Name).To(Equal(bindingName))
-				})
+			It("creates a key binding", func() {
```

Comment:
I would suggesst shortening this `It` to only check the fields of the record that make sense to the `key` binding type, i.e. `Type` being set to `key`, `AppGUID` being empty and `app` relationship not being set. The rest of the fields are being checked in the parent context, so checking them here as well is just unneeded duplication